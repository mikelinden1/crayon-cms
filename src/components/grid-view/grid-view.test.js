import GridView from './grid-view';

const props = {
    items: [
        { id: 1, name: 'item 1' },
        { id: 2, name: 'item 2' },
        { id: 3, name: 'item 3' }
    ],
    config: {
        id: 'widgets',
        moduleName: 'Manage Widgets',
        itemName: 'Widget',
        itemNamePlural: 'Widgets',
        capabilities: {
            reorderable: false,
            editable: true,
            deleteable: true,
            deleteIdProp: 'name',
            bulkImport: true
        },
        itemsPerPage: 6,
        views: {
            displayType: 'grid',
            defaultView: 'grid',
            grid: {
                layout: [
                    { heading: 'Name', name: 'name', gridTag: 'h2' }
                ],
                showId: true
            }
        },
        itemProps: [
            {
                name: 'name',
                label: 'Text',
                type: 'text',
                required: true,
                minLength: 3,
                maxLength: 30
            }
        ]
    }
};

it('GridView matches snapshot', () => {
    const wrapper = shallow(<GridView {...props} />);
    expect(wrapper).toMatchSnapshot();
});