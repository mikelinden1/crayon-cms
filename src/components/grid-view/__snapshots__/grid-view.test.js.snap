// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridView matches snapshot 1`] = `
<div>
  <div
    className="grid-view"
  >
    <Connect(GridViewCard)
      index={0}
      item={
        Object {
          "id": 1,
          "name": "item 1",
        }
      }
      key="grid-card-1"
    />
    <Connect(GridViewCard)
      index={1}
      item={
        Object {
          "id": 2,
          "name": "item 2",
        }
      }
      key="grid-card-2"
    />
    <Connect(GridViewCard)
      index={2}
      item={
        Object {
          "id": 3,
          "name": "item 3",
        }
      }
      key="grid-card-3"
    />
  </div>
</div>
`;
